include/z64save.h

```diff
@@ -74,14 +100,14 @@ typedef struct {
     /* 0x002C */ s16 n64ddFlag;
     /* 0x002E */ s16 healthCapacity; // "max_life"
     /* 0x0030 */ s16 health; // "now_life"
-    /* 0x0032 */ s8 magicLevel;
-    /* 0x0033 */ s8 magic;
+    /* 0x0032 */ s8 magicLevel; // 0 for no magic/new load, 1 for magic, 2 for double magic
+    /* 0x0033 */ s8 magic; // current magic available for use
     /* 0x0034 */ s16 rupees;
     /* 0x0036 */ u16 swordHealth;
     /* 0x0038 */ u16 naviTimer;
-    /* 0x003A */ u8 magicAcquired;
+    /* 0x003A */ u8 isMagicAcquired;
     /* 0x003B */ char unk_3B[0x01];
-    /* 0x003C */ u8 doubleMagic;
+    /* 0x003C */ u8 isDoubleMagicAcquired;
```

Comment:
I didn't realise this was used extensively in the repo, but now I'm wondering if we want to name variables like this interrogatively... does the `is` actually do anything here, that the plain past participle doesn't?

Reply:
I think in this specific instance, `doubleMagicAcquired` alone can be inferred to be a boolean-like value (it still uses the 0/1 in calculating `magicCapacity` so it's not a true bool), but as you mentioned, `is` is just a common and fast way to identify that something is boolean without thinking about it, at least that was my understanding of the convention

Reply:
Also as a side note, not related to this comment but something I wanted to note:
`doubleDefense` -> `isDoubleDefenseAcquired` (or without the `is`) would be consistent with `isDoubleMagicAcquired` but I refrained from changing it for now

Reply:
If we were going by the game it should be "half damage" (or possibly "enhanced defence") anyway:
```
"I'm going to enhance your \n"
COLOR(LIGHTBLUE) "defensive power" COLOR(DEFAULT) "."
[...]
"Your " COLOR(LIGHTBLUE) "defensive power" COLOR(DEFAULT) " is enhanced!"
[...]
"Your defensive power has been \n"
"enhanced! Damage inflicted by \n"
"enemies will be " COLOR(RED) "reduced by half" COLOR(DEFAULT) "."
```

Reply:
imo this is a case where the common colloquial name makes more sense. no one says half damage or enhanced defense

Reply:
I had a look into this, since I was curious. I looked at 4 guidebooks (3 English, 1 the official one, 1 Japanese) and the first page of Google results for walkthroughs, including 4 GameFAQs guides. A large majority say one or more of
- half damage
- enhanced defense
- doubled health
and only 2 (of about 20) say double/doubled defense. I was pretty surprised by this myself, since the tiny speedrunning corner of the world does seem to call always call it "double defense", probably because of the alliteration.

Reply:
Just to add a different perspective, I come from the randomizer community and never heard of any of those terms, it's only ever been "double defense" for me

Reply:
Other related names in decomp: `FAIRY_UPGRADE_HALF_DAMAGE` (hi elliptic), `gGreatFairyDoubleDefenceCs` (hi louis)

I would be okay with normalizing all of that to "half damage", with a comment on the save context struct member like `// double defense`